Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java

```diff
@@ -786,6 +796,13 @@ public String storeXMLQStore(String XML) throws ParserConfigurationException, SA
 		}
 		return res;
 	}
+	
+	private String getAuthHeader(){
+	    UserDetails user = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String auth = user.getUsername() + ":" + user.getPassword();
+        byte[] encodedAuth = Base64.encodeBase64( auth.getBytes(Charset.forName("US-ASCII")) );
+        return "Basic " + new String( encodedAuth );

```

Comment:
you can talk to karthik, he did something similar (you wouldn't need to use spring-security but the properties fiel with pom.xml is the same)